internal/fusefrontend_reverse/reverse_longnames.go

```diff
@@ -51,7 +51,7 @@ func initLongnameCache() {
 // findLongnameParent converts "gocryptfs.longname.XYZ" to the plaintext name
 func (rfs *ReverseFS) findLongnameParent(dir string, dirIV []byte, longname string) (plaintextName string, err error) {
 	longnameCacheLock.Lock()
-	hit := longnameParentCache[longname]
+	hit := longnameParentCache[dir + "/" + longname]
```

Comment:
If you prefer a different solution here, feel free to let me know. I am aware that `dir` is an unencrypted path component, whereas `longname` is an encrypted path component. I still decided to use a concatenated path to avoid a separate struct. The only requirement is that the key is unique for all (dir, longname) pairs.